src/ansys_sphinx_theme/extension/autoapi.py

```diff
@@ -0,0 +1,151 @@
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import os
+import pathlib
+from typing import Any, Dict
+
+from sphinx.application import Sphinx
+
+from ansys_sphinx_theme import __version__
+
+THIS_PARENT_PATH = pathlib.Path(__file__).parent.absolute()
+
+
+AUTOAPI_TEMPLATES_PATH = (
+    THIS_PARENT_PATH.parent.resolve() / "theme" / "ansys_sphinx_theme" / "_templates" / "autoapi"
+)
+
+
+def get_autoapi_templates_dir_relative_path(path: pathlib.Path) -> str:
+    """Return a string representing the relative path for autoapi templates.
+
+    Parameters
+    ----------
+    path : pathlib.Path
+        Path to the desired file.
+
+    Returns
+    -------
+    str
+        A string rerpesenting the relative path to the autoapi templates.
+
+    """
+    print(AUTOAPI_TEMPLATES_PATH.absolute())
+    return os.path.relpath(str(AUTOAPI_TEMPLATES_PATH.absolute()), start=str(path.absolute()))
+
+
+def replace_html_tag(app, exception):
+    """Replace HTML tags in the generated HTML files.
+
+    Parameters
+    ----------
+    app : ~sphinx.application.Sphinx
+        Application instance for rendering the documentation.
+    exception : Exception
+        Exception that occurred during the build process.
+    """
+    if exception is not None:
+        return
+
+    build_dir = pathlib.Path(app.builder.outdir).resolve()
+    if app.config["extensions"] and "autoapi.extension" not in app.config["extensions"]:
+        return
+    api_dir = app.config["autoapi_root"]
+    api_path = build_dir / api_dir
+    if not api_path.exists():
+        return
+    file_names = list(api_path.rglob("*.html"))
+    for file_name in file_names:
+        with open(api_dir / file_name, "r", encoding="utf-8") as file:
+            content = file.read()
+        with open(api_dir / file_name, "w", encoding="utf-8") as file:
+            modified_content = content.replace("&lt;", "<").replace("&gt;", ">")
+            file.write(modified_content)
+
+
+def add_autoapi_theme_option(app: Sphinx) -> None:
+    """Add the autoapi template path to the theme options.
+
+    Parameters
+    ----------
+    app : ~sphinx.application.Sphinx
+        Application instance for rendering the documentation.
+    """
+    autoapi = app.config.html_theme_options.get("autoapi", {})
+    if not autoapi:
+        return
+    required_extensions = ["sphinx_jinja", "sphinx_design"]
+
+    for extension in required_extensions:
+        if extension not in app.config["extensions"]:
+            app.config["extensions"].append(extension)
```

Comment:
Just to be sure, if we add "autoapi" in here, we face the issue in which autoapi complains about the templates directory, right?

Reply:
yes we face the issue of autoapi_dir (path to the src dir)

Reply:
Great! Could we add a `# HACK: ...` comment explaining this so we do not forget the reason?

Reply:
sure, i will add.